### PR TITLE
[FIX] fetchmail: log problems at error level

### PR DESCRIPTION
**Pull request by smortex**
#### Description of the issue/feature this PR addresses:

Fix logging of errors in the fetchmail addon.

#### Current behavior before PR:

Error logged by the fetchmail addons are logged at 'info' level.

#### Desired behavior after PR is merged:

Error logged by the fetchmail addons are logged at 'error' level.

----

- [X] I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by smortex** [comment]
Hu… looks like I should be targetting another branch (since 13.0 also has this problem and older branches too), but editing this through GitHub seems to do the wrong thing :-/  Should I close this PR and setup a new one with the correct base branch?

**Comment by xmo-odoo** [comment]
@smortex looks like this is reverting a change specifically made by @simongoffin way back when (0fd773a486fbc1aeef7f00ec5eab3eaf0e4c6a9d), you may want to ask them if they remember why these notifications were downgraded.

Also `logging.error(..., exc_info=True)` has a built-in version: `logging.exception`.

**Comment by mart-e** [comment]
Also, if accepted, this should target the master branch not the end-of-life 12.0 that should only receive critical bugfixes.
Could you detail a bit more why you need this change?

File: addons/fetchmail/models/fetchmail.py [hunk 1]
```diff
@@ -130,7 +130,7 @@ def button_confirm_login(self):
                 connection = server.connect()
                 server.write({'state': 'done'})
             except Exception as err:
-                _logger.info("Failed to connect to %s server %s.", server.type, server.name, exc_info=True)
+                _logger.error("Failed to connect to %s server %s.", server.type, server.name, exc_info=True)
                 raise UserError(_("Connection test failed: %s") % tools.ustr(err))
```
Comment (smortex) [inline addons/fetchmail/models/fetchmail.py:134]:
```suggestion
                raise RuntimeError(_("Connection test failed: %s") % tools.ustr(err))
```

For consistency we should probably also change the class of this exception

**Review by smortex — COMMENTED** [review]
Good catch @xmo-odoo, I am looking forward for feedback from @simongoffin :+1: 

@mart-e I discovered this issue unintentionally on one of our sites: the e-mail account password has been changed at the internet service provider level, and we only consider warning and above as "interesting" things to look after in our logs, so we missed this for some time.

I guess that at the time a user setup an account, errors can be considered UserError. but once it's setup and validated, any error happening is more interesting for the administrator of the system running odoo than the user who configured it.  So I think that this better fit in the "warning" or "error" levels from the description of the commit 0fd773a486fbc1aeef7f00ec5eab3eaf0e4c6a9d:

> This commit introduces a more consistent policy for logging errors and warnings:
>  - All messages that do not require administrator attention should be logged at INFO level or lower. This includes all errors that are notified to the user in a friendly manner, even for access right problems or validation errors during business operations.
>  - All messages that indicate a likely misconfiguration or malicious use by the users should be logged at WARNING level, as they typically require administrator attention.
>  - All other unhandled internal errors cannot typically be handled by the user and should be logged at ERROR or higher level, as they require immediate administrator attention.

I am therefore targetting the `12.0` branch because all supported releases are affected, and I would like this fix to be applied to all of them. I understood that I should target the oldest branch that should have the fix…

**Comment by C3POdoo** [comment]
Dear @smortex,

Thank you for your contribution but the version 12.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**